package Persistent.Proto;

message Term {
  optional int64 id = 1;
  optional string name = 2;
}

message Entry {
  optional int64 id = 1;
  optional string author = 2;
  optional string text = 3;
  optional string date_added = 4;
  optional int64 term_id = 5;
}

message CreateTermRequest {
  optional string name = 1;
}

message CreateTermResponse {
  optional Term term = 1;
}

message GetTermRequest {
  optional string name = 1;
}

message GetTermResponse {
  optional Term term = 1;
}

message CreateEntryRequest {
  optional Term term = 1;
  optional string author = 2;
  optional string text = 3;
}

message CreateEntryResponse {
  optional Entry entry = 1;
}

message GetEntriesRequest {
  optional Term term = 1;
}

message GetEntriesResponse {
  repeated Entry entries = 1;
}

service PersistentService {
  rpc CreateTerm(CreateTermRequest)
    returns (CreateTermResponse);
  rpc GetTerm(GetTermRequest)
    returns (GetTermResponse);
  rpc CreateEntry(CreateEntryRequest)
    returns (CreateEntryResponse);
  rpc GetEntries(GetEntriesRequest)
    returns (GetEntriesResponse);
}